zad2/warriors.rb

```diff
@@ -1,12 +1,95 @@
-class Warrior
-  def initialize(name:, level:)
+class Character
+  attr_accessor :name, :level
+
+  def initialize name, level
+    self.name = name
+    self.level = level
+  end
+
+  def card
+  puts "Character's name: #{name}"
+  puts "Character's level: #{level}"
   end
+end
+
+class Warrior < Character
+
+def initialize (name, level)
+  @name = name
+  @level = level
+end
 
   def strength
+  @level + [*1..12].sample
   end
+
+def level_up(looser)
+      lev = self.level >= looser.level.to_i ? 1 : looser.level.to_i - self.level.to_i
+      self.level += lev.to_i
+  end
+end
+
+class Monster < Character
+
+def initialize (name, level)
```

Comment:
nie potrzeba `initialize` w `Warrior` / `Monster`, jeśli klasy te dziedziczą po `Character` posiadającą tę metodę i wykonującą dokładnie to samo